backend/services/UserService.js

```diff
@@ -1,39 +1,49 @@
-const BaseModelService = require('./baseModelService');
+const BaseModelService = require('./BaseModelService');
 
 class UserService extends BaseModelService {
     constructor(bcryptjs) {
         super();
         this.bcryptjs = bcryptjs;
     }
 
-    static get table() {
+    get table() {
         return 'users';
     }
 
-    static register(email, password) {
+    register(email, password) {
         const salt = this.bcryptjs.genSaltSync(10);
         const encryptedPassword = this.bcryptjs.hashSync(password, salt);
 
         return this.create({
             email,
             password: encryptedPassword,
-            created_at: Date.now()
+            createdAt: Date.now()
         });
     }
 
-    static async login(email, password) {
+    async login(email, password) {
         const user = await this.findOne({ email });
 
         if (!user) {
             return null;
         }
 
         if (this.bcryptjs.compareSync(password, user.password)) {
-            return this.findOne({ email });
+            return this.getById(user.id);
         }
 
         return null;
     }
+
+    async getById(id, excludePassword = true) {
+        const result = await this.couch.get(this.table, id);
+
+        if (excludePassword) {
+            delete result?.data?.password;
+        }
+
+        return result.data;
```

Comment:
something is wrong there, because above you are checking if `data` exists in `result` and there you arent, so I guess there can be `can not read porperty data of undefined or null`

Reply:
Dawid, check this 'data' I think it's always exists but sometimes is null. 

So we can do:
const { data } = 

if (!data) {
   return null;
}

if () { delete data.password; }

return data;